src/MPDevice.cpp

```diff
@@ -4443,42 +4460,42 @@ void MPDevice::changeVirtualAddressesToFreeAddresses(void)
 {
     if (virtualStartNode != 0)
     {
-        qDebug() << "Setting start node to " << freeAddresses[virtualStartNode].toHex();
-        startNode = freeAddresses[virtualStartNode];
+        qDebug() << "Setting start node to " << getFreeAddress(virtualStartNode).toHex();
```

Comment:
how do you handle knowing if you're getting a free address for a parent node or child node, as they have different sizes?